src/poetry/core/semver/version_union.py

```diff
@@ -302,26 +310,31 @@ def _excludes_single_wildcard_range_check_is_valid_range(
         wildcard range.
         """
 
-        max_precision = max(one.max.precision, two.min.precision)
+        one_max = one.max
+        assert one_max is not None
+        two_min = two.min
+        assert two_min is not None
+
```

Comment:
Can't we just replace this and avoid the reassignment?

```suggestion
        assert one.max is not None
        assert two.min is not None

```

Reply:
huh, I'm slightly surprised to find that you can.  Not sure how mypy is deciding that what is true of `one.max` at some point is necessarily also true at some later point, but it doesn't complain.